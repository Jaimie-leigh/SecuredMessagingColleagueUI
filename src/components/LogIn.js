// Inital screen displayed on load on the Colleague Interface

import React, { useState } from "react";
import { useHistory } from "react-router-dom";
import "./Login.css";

export default function LogIn() {
  const history = useHistory();
  const [value, setValue] = useState("");
  const regexp = /^[0-9]{0,7}$/;
  const [isValid, setIsValid] = useState("");
  
  //sets value from username entered by user
  // sets isValid to true or false if username is in expected format using regular expression 
  function handleChange(e) {
    setValue(e.target.value);
    if (!regexp.test(value)) {
      setIsValid(false);
    } else {
      setIsValid(true);
    }
  }

  // on click of 'Log in' button checks username is in correct format
  // routes to colleague landing page if correct 
  function handleSubmit(e) {
    e.preventDefault();
    if(!isValid)
    {
      return
    }
    return history.replace("/ColleagueWelcome", value);
  }

  // renders on the page
  return (
    <div className="Login">
      <form onSubmit={handleSubmit}>
        <h1>Secured Messenging</h1>
        <label>
          Username:
          <input
            type="tel"
            name="ColleagueId"
            value={value}
            onChange={handleChange}
            required
            maxLength="6"
            minLength="6"
          />
        </label>
        {isValid === false && (
          <div className="helpText">
            <p>Username should consist of 6 characters including only digits, please re-enter your username</p>
          </div>
        )}
        <button>Log In</button>
      </form>
    </div>
  );
}
