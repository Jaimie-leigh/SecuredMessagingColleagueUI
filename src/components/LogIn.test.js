import React, { useState } from 'react';
import renderer from 'react-test-renderer';
import {render, unmountComponentAtNode } from 'react-dom';
import LogIn from './LogIn';
import {fireEvent, screen, queryByLabelText } from '@testing-library/react'
//import { screen, getByLabelText } from '@testing-library/dom'

let container = null;

beforeEach(() => {
    // set a DOM elemant as render target
    container = document.createElement('div');
    document.body.appendChild(container);
})

afterEach(() => {
    // clean up on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
})

describe('Log in snapshot test', () => {
    test('test renders as per snapshot', () => {
        const tree = renderer
            .create(<LogIn></LogIn>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    // test("test page renders as epxected when invalid text entered too userID field", () => {
    //     const searchInput = screen.queryByLabelText('ColleagueId');
    
    //     render(
    //       <LogIn></LogIn>,
    //       container
    //     );

    //     userEvent.type(screen.queryByLabelText('ColleagueId'), 'abcde')
    //     //expect(screen.getByRole('textbox')).toHaveValue('Hello,\nWorld!')

    //     // fireEvent.change(searchInput, { target: { value: 'abcde' }})
    //      expect(container.innerHTML).toContain("numbers only allowed");
    //   });
});
