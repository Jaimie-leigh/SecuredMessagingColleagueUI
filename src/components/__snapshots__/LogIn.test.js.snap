// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Log in snapshot test test renders as per snapshot 1`] = `
<div
  className="Login"
>
  <form
    onSubmit={[Function]}
  >
    <h1>
      Secured Messenging
    </h1>
    <label>
      Username:
      <input
        maxLength="6"
        minLength="6"
        name="ColleagueId"
        onChange={[Function]}
        required={true}
        type="tel"
        value=""
      />
    </label>
    <button>
      Log In
    </button>
  </form>
</div>
`;
